chipy_org/apps/meetings/models.py

```diff
@@ -229,7 +233,80 @@ def __str__(self):
     start_time = models.DateTimeField(blank=True, null=True)
     approved = models.BooleanField(default=False)
 
-    objects = TopicsQuerySet.as_manager()
+    objects = TopicQuerySet.as_manager()
+
+    def outstanding(self):
+        return self.drafts.filter(
+            models.Q(topic__meeting__when__gte=timezone.now()) | models.Q(topic__meeting=None)
+        ).filter(approved=False)
+
+    class Meta:
+        abstract = True
+
+    def __eq__(self, other):
+        for fld in TopicDraft.tracked_fields:
+            if getattr(self, fld) != getattr(other, fld):
+                return False
+        return True
+
+
+class Topic(TopicBase):
+    pass
+
+
+class TopicDraftQuerySet(models.QuerySet):
+    def get_user_drafts(self, user):
+        return self.filter(topic__presentors__user=user)
+
+
+class TopicDraft(TopicBase):
+    topic = models.ForeignKey("meetings.Topic", on_delete=models.CASCADE, related_name="drafts")
+
+    tracked_fields = [
+        "title",
+        "meeting",
+        "experience_level",
+        "length_minutes",
+        "description",
+        "slides_link",
+    ]
+
+    objects = TopicDraftQuerySet.as_manager()
+
+    def _copy_tracked(self, src, dst):
+        for fld in TopicDraft.tracked_fields:
+            setattr(dst, fld, getattr(src, fld))
+        dst.save()
+
+    def publish(self):
+        topic = self.topic
+        orig_notes = topic.notes or ""
+        new_notes = self.notes or ""
+        topic.notes = (
+            orig_notes
+            + "\n----------------------------------------------\n"
+            + "Published Draft {} on {}\n".format(self.id, timezone.now())
+            + new_notes
+            + "\n----------------------------------------------"
+        )
+        self._copy_tracked(self, topic)
+
+    def __rrshift__(self, other):
+        self.topic = other
+        self._copy_tracked(other, self)
+
+    def __eq__(self, other):
+        for fld in TopicDraft.tracked_fields:
+            if getattr(self, fld) != getattr(other, fld):
+                return False
+        if other.id != self.topic.id:
+            return False
+        return True
+
+    def __hash__(self):
+        if self.pk is None:
+            raise TypeError("Model instances without primary key value are unhashable")
+        return hash(self.pk)
```

Comment:
I think this is causing an error as it's the only hash function implemented / changed here.  I am unable to delete topics through the admin.  

![image](https://user-images.githubusercontent.com/3450848/82445040-c70d4280-9a69-11ea-9390-317faf689710.png)

followed by:

![image](https://user-images.githubusercontent.com/3450848/82445111-e3a97a80-9a69-11ea-84e4-65a1fcb15c2c.png)

Reply:
Also, I did confirm this functionality does work on master.

Reply:
Also isn't this the implementation the default?  does it need to be overridden specifically?  

[django model hash](https://docs.djangoproject.com/en/2.2/ref/models/instances/#hash)

Reply:
[code](https://docs.djangoproject.com/en/2.2/_modules/django/db/models/base/#Model.__hash__)